src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonNode.cs

```diff
@@ -50,12 +51,14 @@ internal JsonNode(JsonNodeOptions? options = null)
         /// </exception>
         public JsonArray AsArray()
         {
-            if (this is JsonArray jArray)
+            JsonArray? jArray = this as JsonArray;
+
+            if (jArray is null)
```

Comment:
```suggestion
            if (this is not JsonArray jArray)
```

Reply:
Hi @eiriktsarpalis  I'm getting this error for your suggestion:
![ảnh](https://github.com/dotnet/runtime/assets/11854062/28cd83e8-b332-4dcb-8542-6509f35c1118)
Can you help me with it ? try to use debug not null assertion but still doesn't work.

Reply:
Ah yes. I don't think we can fix this without adding `return null!;` statement at that branch but that's adding unnecessary statements. Let's keep as-is.